hatch/src/cli/commands/build.rs

```diff
@@ -18,37 +18,34 @@ impl<'build> Build {
   }
 
   pub fn execute(&self, project: &Project) -> HatchResult<()> {
-    task::generate_assets(&project).with_context(|e| {
-      format!("Failed to generate assets : {}", e)
-    })?;
-
-    if let Some(path) = project.path().to_str() {
-      let command = format!("cd {} && tup", path);
-      match task::platform_type() {
-        PlatformKind::Windows => {
-          let mut child =
-            process::Command::new("cmd")
-              .arg("/C")
-              .arg(command)
-              .spawn().with_context(|e| {
-              format!("failed to build project at `{}` : {}", &path, e)
-            })?;
-          child.wait()?;
-        },
-        _ => {
-          let mut child =
-            process::Command::new("sh")
-              .arg("-c")
-              .arg(command)
-              .spawn().with_context(|e| {
-              format!("failed to build project at `{}` : {}", &path, e)
-            })?;
-          child.wait()?;
+    let res = (|| -> HatchResult<()> {
```

Comment:
Can we move this error wrap into the method below it?

Reply:
The idea is that I don't want to create the failed to build error in every call to execute.  This is essentially building the error message in so that we don't have a bunch of these context statements everywhere repeating ourselves, just like you wanted :).

Reply:
Huh? Thats exactly where you want it, its the top of the error propagation.

Reply:
Hmm this actually an interesting issue.  We want a similar mechanism in both places because we call the top level execute and the Build::execute method.  I could move it to the top level, but I would have to repeat myself a bit when handling the direct call to Build::execute.

Reply:
Well you would not put it in the Build::execute method, you would put it in both Command::execute methods of test and build